File: exercism/cpp/grains/grains.cpp
#include "grains.h"

namespace grains
{

    unsigned long long square(unsigned int square)
    {
        return 1ULL << (square - 1);
    }
    unsigned long long total()
    {
        unsigned long long t = 0;
        for (unsigned int i = 1; i <= 64; i++)
        {
            t+= square(i);
        }
        return t;   
    }
} // namespace grains

File: exercism/julia/grains/grains.jl
"""Calculate the number of grains on square `square`."""
function on_square(square)
    if 1 <= square <= 64
        2 ^ (big(square) - 1)
    else
        throw(DomainError(square, "$square not in [1:64]"))
    end
end

"""Calculate the total number of grains after square `square`."""
function total_after(square)
    if 1 <= square <= 64
        (2 ^ big(square)) - 1
    else
        throw(DomainError(square, "$square not in [1:64]"))
    end
end
